.github/workflows/freebsd.yml

```diff
@@ -0,0 +1,61 @@
+# This file is a part of the UMSKT Project
+#
+# Copyleft (C) 2019-2023 UMSKT Contributors (et.al.)
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# @FileCreated by techguy16 on 07/23/2023
+# @Maintainer techguy16
+
+name: C/C++ CI (FreeBSD)
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+    runs-on: macos-12
```

Comment:
why is this running on macos?

Reply:
the macos runner appears to have virtualbox installed by default, which is the easiest way to compile for freebsd it appears

Reply:
ah wait, I've found a better solution
https://github.com/marketplace/actions/cross-platform-action

Reply:
No the macOS + VB is better.